department-of-veterans-affairs/vets-api
# Add logout and revoke all sessions endpoint

## Summary
- Logout and revoke all sessions for SiS. This will allow a user to complete the logout flow and cleanup to login.gov session.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#67750

## How to test
- In `vets-website` change [this line](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/utilities/oauth/utilities.js#L297C1-L297C1) to:
```js
API_SIGN_IN_SERVICE_URL({ endpoint: 'logout_and_revoke_all_sessions' }),
```
- start `vets-website`
- start `vets-api`
- in browser `#1` login with SiS and accept terms
- in browser `#2` login with SiS
- in browser `#1` navigate to [localhost:3001/terms-of-use](localhost:3001/terms-of-use) and Decline.
- You should be logged out of browser `#1`.
- In browser `#2` you should be logged out once you access token expires (~5 mins)
- Repeat for login.gov (and a combination of logingov/idme)
   - When you logout of browser `#1` using login gov you should be redirected to login.govs logout page

## What areas of the site does it impact?
Sign-in Service

## Acceptance criteria
